planet_scope/cloudless_mosaic/script.js

```diff
@@ -0,0 +1,66 @@
+//VERSION=3
+//Cloudless Mosaic with PlanetScope
+
+function setup() {
+  return {
+    input: ["red", "green", "blue", "dataMask", "clear"],
+    output: { bands: 4 },
+    mosaicking: "ORBIT"
+  };
+}
+
+function preProcessScenes(collections) {
+  collections.scenes.orbits = collections.scenes.orbits.filter(function (orbit) {
+    var orbitDateFrom = new Date(orbit.dateFrom);
+    return orbitDateFrom.getTime() >= (collections.to.getTime() - 7 * 24 * 3600 * 1000);
+  });
+  return collections;
+}
+
+function getLastObservation(arr) {
+  for (let i = arr.length - 1; i >= 0; i--) {
+    if (arr[i] !== 0) { // optional check if you are sure all invalid observations are filtered out
+      return arr[i];
+    }
+  }
+  return 0;
+}
+
+function getMedian(sortedValues) {
+  var index = Math.floor(sortedValues.length / 2);
+  return sortedValues[index];
+}
+
+function evaluatePixel(samples, scenes) {
+  var reds = []; var greens = []; var blues = []; //empty arrays for reds greens and blues
+  var a = 0; //incrementer
+
+  for (var i = 0; i < samples.length; i++) { //for each sample
+    var sample = samples[i]; //get current sample
+    var clear = 0 + (sample.dataMask * sample.clear); //0 for clouds OR datamask, 1 for neither
```

Comment:
Might be clearer with logic operators:

```suggestion
    var clear = sample.dataMask && sample.clear; //0 for clouds OR datamask, 1 for neither
```